xdsl/dialects/arm/ops.py

```diff
@@ -18,14 +19,51 @@ class ARMOperation(IRDLOperation, ABC):
     Base class for operations that can be a part of ARM assembly printing.
     """
 
+    @abstractmethod
+    def assembly_line(self) -> str | None:
+        raise NotImplementedError()
+
+
+class ARMInstruction(ARMOperation, ABC):
```

Comment:
Will `ARMOperation`s that are not `ARMInstructions` ever print assembly? If not then I imagine that the `assembly_line` method can be removed from `ARMOperation`

Reply:
Yep! The `get_register` operation participates in assembly printing by explicitly saying that there's nothing to print. There will be more like this.

Reply:
I'm not sure I understand, is explicitly saying there's nothing to print different to printing nothing? Another way to phrase the question above is:
Will there be any `ARMOperation` which is not an `ARMInstruction for which `assembly_line` does not return `None`

Reply:
In RISC-V, examples of this are CommentOp, and RiscvFunc.FuncOp